### PR TITLE
OOP school library: add basic UI

### PR DESCRIPTION
**Pull request by kinginthenorthcodez**
## OOP school library: add basic UI

Build interactive console apps.
Run a program using the command line.

- Your console app should behave in the same way is in the example.
- Create a app.rb file that will serve as your console app entry-point. It should have methods that do the following:
List all books.
List all people.
Create a person (teacher or student, not a plain Person).
Create a book.
Create a rental.
List all rentals for a given person id.
- In your main.rb define the entry point, this will be a method called main that is invoked at the end of your file. This method should do the following:
Present the user with a list of options to perform.
Lets users choose an option.
If needed, ask for parameters for the option.
Have a way to quit the app.

Warm Regards
@kinginthenorthcodez

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+    loop do
+      print 'Enter ID of person from listed persons'
+      id = parse_int
+      persons.each do |person|
+        if person.id == id
+          selcted_person = person
+          break
+        end
+      end
+      next unless selcted_person
```
Comment (DeliceLydia) [inline app.rb]: - [ ] I am getting an error when I try to List all rentals for a given id'

File: app.rb [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+      print "\n"
+      print 'Age: '
+      age = gets.chomp
+      print "\n"
+      print 'specialization: '
+      specialization = gets.chomp
+      print "\n"
+      age = age.to_i
+      break if name.length.positive? && specialization.length.positive? && age.positive?
+
```
Comment (DeliceLydia) [inline app.rb:45]: - [ ] I am not able to create a `teacher` when I try to add `specialization` I get an error please address that
gg

**Review by DeliceLydia — CHANGES_REQUESTED** [review]
## Hi @kinginthenorthcodez, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Required Changes ♻️
_Check the comments under the review._



### Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)

File: app.rb (hunk 1 above)
Comment (kinginthenorthcodez) [reply app.rb]: Thanks @DeliceLydia  required changes made 👍🏽 
image

File: app.rb (hunk 2 above)
Comment (kinginthenorthcodez) [reply app.rb:45]: @DeliceLydia  addressed the issue 👍🏽 

image

**Review by Ismailco — CHANGES_REQUESTED** [review]
### Hi @kinginthenorthcodez, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights

✔️  No linter errors
✔️ Used Gitflow
✔️ Work documented professionally

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app.rb [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+      puts 'Wrong input: Try again *'
+    end
+    person = persons[person_index - 1]
+    book = books[book_index - 1]
+    rental = Rental.new(date, person, book)
+    rentals.push(rental)
+  end
+
+  def list_rentals
```
Comment (Ismailco) [inline app.rb:140]: - [ ] As the previous reviewer mentioned, the rentals returned an error and break the app after providing the person's id

Screenshot_20220730_171807

**Review by Ismailco — CHANGES_REQUESTED** [review]

File: app.rb (hunk 3 above)
Comment (kinginthenorthcodez) [reply app.rb:140]: @Ismailco resolved issue thanks 👍🏽 
image

**Review by ahmetbozaci — APPROVED** [review]
# 🥳 🎉🥇 💯  💥 🎆 🎈 STATUS: APPROVED 🥳 🎉  🥇  💯  💥   🎆 🎈 🎉 



@ ✋🏼 
#### Your project is excellent. 🎉 I approve of it. There is nothing else to say other than... it's time to merge it. You can submit this project on your dashboard.
___ 

### 💯 Highlights 💯
- I can show rentals with person ıd without an error. ✔️ 

___ 

### Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
___

### Cheers and Happy coding! 💻 👏
aproved


___

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification._
___

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form.](https://airtable.com/shrQAqnBwek5a0O0s)_

[GitHub](https://github.com/ahmetbozaci) | [LinkedIn](https://linkedin.com/in/ahmetbozaci) | [Twitter](https://twitter.com/ahmtbozaci)

_Reviewed by [Ahmet Bozacı ](https://github.com/ahmetbozaci)_

**Comment by kinginthenorthcodez** [comment]
Merged #4  into development

**Pull request merged**